File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.js
/**
 * @param {number[]} nums
 * @param {number[]} queries
 * @return {number[]}
 */
var answerQueries = function(nums, queries) {
    let answer = []
    nums.sort((a,b) => a-b) 
    for(let  i= 0; i<queries.length; i++){
        let sum=0;
        let count=0;
        for(let j=0 ; j<nums.length; j++){
            if(sum+nums[j]<=queries[i]){
                sum+=nums[j];
                count++;
            }else break;
        }
        answer.push(count)
    }
    return answer
};

File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.cpp
class Solution
{
public:
    vector<int> answerQueries(vector<int> &nums, vector<int> &queries)
    {
        vector<int> answer;
        sort(nums.begin(), nums.end());
        for (int i = 0; i < queries.size(); i++)
        {
            int count = 0;
            int sum = 0;
            for (int j = 0; j < nums.size(); j++)
            {
                if (sum + nums[j] <= queries[i])
                {
                    sum += nums[j];
                    count++;
                }
                else
                {
                    break;
                }
            }
            answer.push_back(count);
        }
        return answer;
    }
};